docs/docs/adding-search-with-algolia.md

```diff
@@ -2,593 +2,582 @@
+
+```jsx:title=src/components/search/search-result.js
+import { Link } from "gatsby"
+import { default as React } from "react"
+import {
+  connectStateResults,
+  Highlight,
+  Hits,
+  Index,
+  Snippet,
+  PoweredBy,
+} from "react-instantsearch-dom"
+
+const HitCount = connectStateResults(({ searchResults }) => {
+  const hitCount = searchResults && searchResults.nbHits
+
+  return hitCount > 0 ? (
+    <div className="HitCount">
+      {hitCount} result{hitCount !== 1 ? `s` : ``}
+    </div>
+  ) : null
+})
+
+const PageHit = ({ hit }) => (
+  <div>
+    <Link to={hit.slug}>
+      <h4>
+        <Highlight attribute="title" hit={hit} tagName="mark" />
+      </h4>
+    </Link>
+    <Snippet attribute="excerpt" hit={hit} tagName="mark" />
+  </div>
+)
+
+const HitsInIndex = ({ index }) => (
+  <Index indexName={index.name}>
+    <HitCount />
+    <Hits className="Hits" hitComponent={PageHit} />
+  </Index>
 )
 
-const Stats = connectStateResults(
-  ({ searchResults: res }) =>
-    res && res.nbHits > 0 && `${res.nbHits} result${res.nbHits > 1 ? `s` : ``}`
+const SearchResult = ({ indices, className }) => (
+  <div className={className}>
+    {indices.map(index => (
+      <HitsInIndex index={index} key={index.name} />
+    ))}
+    <PoweredBy />
+  </div>
 )
+
+export default SearchResult
 ```
 
-Now comes the actual `Search` component. It starts off with some state initialization, defining handler functions and event listeners to trigger them. All they do is make the search input slide out when the user clicks a search icon and disappear again when the user clicks or touches (on mobile) anywhere.
+Since Algolia supports multiple indices, the `SearchResult` iterates over all indices and displays hits for each of them using the `HitsInIndex` component. It, in turn, relies heavily on the [`Hits` component](https://www.algolia.com/doc/api-reference/widgets/hits/react/) from the InstantSearch library.
+
+The `PageHit` component is responsible for displaying a single page ("hit") in a search result.
+
+[`connectStateResults`](https://community.algolia.com/react-instantsearch/connectors/connectStateResults.html) wraps components to provide them with details about the current search such as the query, the number of results and timing statistics.
+
+If you're using Algolia's free tier, they ask you to acknowledge the use of their technology by including the string "Powered by Algolia", which is what `PoweredBy` does.
+
+`Highlight` and `Snippet` both display attributes of matching search results to the user. The former renders the full value whereas the latter only shows a snippet. A snippet is the text immediately surrounding the match. The `attribute` property is the name of the key in the Algolia index (as generated by `pageToAlgoliaRecord` in `algolia-queries.js`).
+
+### Tying the search widget together
+
+You now need to hook up the two components to each other and perform the actual search:
 
 ```jsx:title=src/components/search/index.js
```

Comment:
`search/index.js` has already been defined in this project, should this have a different name? Ideally there should be minimal revisiting of the same file in a reference guide.

Reply:
No worries!